File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> CharSet = new HashSet<>();

        int left = 0;
        int result = 0;

        for(int r = 0; r < s.length() ; r++){
            while(CharSet.contains(s.charAt(r))){
                CharSet.remove(s.charAt(left));
                left++;
            }
            CharSet.add(s.charAt(r));
            result = Math.max(result , r-left +1);
        }

        return result;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        charSet = set()
        l = 0
        res =0

        for _ in range(len(s)):
            while s[_] in charSet:
                charSet.remove(s[l])
                l +=1
            charSet.add(s[_])
            res= max(res , _-l +1)
        
        return res
        